### PR TITLE
Replace custom Marvin copy with string.GetHashCode

### PR DESCRIPTION
**Pull request by stephentoub**
There's no `ReadOnlySpan<byte>`-based hashcode routine, so in both cases this roundtrips through `ReadOnlySpan<char>` in order to use string's.

### CODE REVIEW

**Comment by GrabYourPitchforks** [comment]
Heads up that most of our span-consuming APIs - including `string.GetHashCode` - have undefined behavior when given misaligned refs.

**Comment by stephentoub** [comment]
> Heads up that most of our span-consuming APIs - including string.GetHashCode - have undefined behavior when given misaligned refs.

Undefined meaning they don't work today on all supported platforms, or meaning we won't promise they'll work in the future?

I can replace some or all of the usage with HashCode as needed; it's a bit slower, but functional.

**Comment by GrabYourPitchforks** [comment]
> Undefined meaning they don't work today on all supported platforms, or meaning we won't promise they'll work in the future?

With very specific exceptions, they're explicitly not tested for this scenario today even on supported platforms, there are no guarantees about future behavior. See also https://github.com/dotnet/runtime/issues/44497.

IMO a better course of action would be to add `ROS<byte>`-consuming APIs to `System.HashCode`. It could be used both by this and by a bunch of other scenarios, such as if we ever need to add randomized hash code calculation to other primitives like `double`.

**Comment by stephentoub** [comment]
> IMO a better course of action would be to add ROS<byte>-consuming APIs to System.HashCode. It could be used both by this and by a bunch of other scenarios, such as if we ever need to add randomized hash code calculation to other primitives like double.

@eiriktsarpalis and I were discussing that earlier today.  He was going to submit an API proposal.

**Comment by stephentoub** [comment]
I changed it to just use HashCode.Combine/Add.

**Comment by stephentoub** [comment]
@GrabYourPitchforks, mind re-reviewing?

File: src/libraries/System.Formats.Cbor/src/System/Formats/Cbor/CborConformanceLevel.cs [hunk 1]
```diff
@@ -209,7 +210,20 @@ public static bool RequireCanonicalSimpleValueEncodings(CborConformanceMode conf
 
         public static int GetKeyEncodingHashCode(ReadOnlySpan<byte> encoding)
         {
-            return System.Marvin.ComputeHash32(encoding, System.Marvin.DefaultSeed);
+            HashCode hash = default;
+
+            while (encoding.Length >= sizeof(int))
```
Comment (stephentoub) [inline src/libraries/System.Formats.Cbor/src/System/Formats/Cbor/CborConformanceLevel.cs:215]:
```suggestion
            // TODO: Use https://github.com/dotnet/runtime/issues/48702 if/when it's available
            while (encoding.Length >= sizeof(int))
```

**Comment by stephentoub** [comment]
@GrabYourPitchforks, can you take another look?

**Comment by GrabYourPitchforks** [comment]
LGTM, by the way!

**Comment by stephentoub** [comment]
Thanks :)  Nothing like merging to get someone's attention :wink:

**Pull request merged**